### PR TITLE
Implement Zaken API 1.2 features

### LINKED ISSUE #1218: As user of Open Zaak, I want Open Zaak to support Zaken API 1.2
... so I can use the features described in https://github.com/VNG-Realisatie/zaken-api/blob/master/CHANGELOG.rst#120-2021-08-27

**Description**
Features from zaken API 1.2 should be added

Features:
- [x] ZaakObject put/patch/delete ([ref PR](https://github.com/VNG-Realisatie/zaken-api/pull/202))
  - https://github.com/open-zaak/open-zaak/pull/1221
- [x] ZaakEigenschap put/patch/delete ([ref PR](https://github.com/VNG-Realisatie/zaken-api/pull/195))
  - https://github.com/open-zaak/open-zaak/pull/1220
- [x] Zaakobjecten overige definitie ([ref PR](https://github.com/VNG-Realisatie/zaken-api/pull/203))
  - https://github.com/open-zaak/open-zaak/pull/1222

TODO:
- [x] Increment Zaken API version to 1.2 where needed

### PR DESCRIPTION
**Pull request by sergei-maertens**
Fixes #1218

**Changes**

* ZaakObject.objectTypeOverigeDefinitie
* ZaakObject writable
* ZaakEigenschap writable

### CODE REVIEW

**Comment by sergei-maertens** [comment]
@stevenbal I've checked the schema in this branch against the upstream schema diff for 1.2.0 and this looks like it's complete :tada:

File: src/openzaak/components/documenten/openapi.yaml [hunk 1]
```diff
@@ -2056,7 +2056,7 @@ components:
           maxLength: 200
           minLength: 1
         locked:
-          title: locked
+          title: geblokkeerd
```
Comment (stevenbal) [inline src/openzaak/components/documenten/openapi.yaml:2059]: Was this change intentional? It seems that the Dutch description still says "gelocked"

File: src/openzaak/components/zaken/api/viewsets.py [hunk 2]
```diff
@@ -614,12 +649,11 @@ def perform_destroy(self, instance):
 
 @conditional_retrieve()
 class ZaakEigenschapViewSet(
-    NotificationCreateMixin,
+    NotificationViewSetMixin,
     AuditTrailCreateMixin,
```
Comment (stevenbal) [inline src/openzaak/components/zaken/api/viewsets.py:653]: Shouldn't this also use `AuditTrailViewSetMixin`?

File: src/openzaak/components/documenten/openapi.yaml (hunk 1 above)
Comment (sergei-maertens) [reply src/openzaak/components/documenten/openapi.yaml:2059]: yeah not sure what's happening here, but re-generating the schema doesn't change the value. Looking at the po file, this does seem intentional:

```po
#: src/openzaak/components/documenten/admin.py:279
#: src/openzaak/components/documenten/api/serializers.py:226
msgid "locked"
msgstr "geblokkeerd"
```

**Review by stevenbal — APPROVED** [review]
